CHECKLIST.md

```diff
@@ -531,11 +531,11 @@ None. No matter how challenging it feels to avoid implementation types in public
 
 [Simplicity](https://theamiableapi.com/2011/09/07/api-design-best-practice-keep-it-simple/) and [Consistency](https://theamiableapi.com/2011/09/14/api-design-best-practice-consistency/). Each API deserves its own package, so that developers can focus on the features they need, without any distracting clutter. The most frequently heard argument for putting several APIs into the same package is their small size. This argument fails to consider that APIs evolve, and what is a small package today can easily become a very large package containing a hodge-podge of unrelated features over time. The java.util package is the best know example.
 
-### 1.1.10.  **Do not** place API and SPI into the same package
+### 1.1.10.  **Avoid** relying on documentation to control evolution of API or SPI
```

Comment:
@dagguh Why is it in conflict with the previous version? 

Maybe **Do not** was too strong, it could be changed to **Consider**. And we may work on the rationale. Putting inside of a different package doesn't mean you need to have strict contracts (API definition) that separate the concerns. It may be a way to optimise a user learning process. It's easier to start with a library which shows less API at the beginning. In most cases, SPI is dedicated to advanced users.

Reply:
I felt it's in conflict, because each SPI is automatically an API and therefore they're always mixed.
OTOH not all API is SPI, so maybe the split is about that.
However, 1.1.10 is the only rule mentioning SPI. SPI is not well defined here. I'd use the [Occam's razor](https://en.wikipedia.org/wiki/Occam%27s_razor) and remove the SPI term altogether. Fewer terms, less complexity, easier rules.

Reply:
> There always need to be the first one.

Sure, on its own it's not enough to reject SPI concept. However, it's a strong hint, that it might not be necessary to exist.
Let's evaluate the pros and cons. So far I see this:

Pros:
* some already talk about SPI

Cons:
* additional mechanism - complexity, mental cost
* unnecessary - we can talk about interfaces and final classes without API/SPI dichotomy